processing/src/main/java/io/druid/query/groupby/GroupByQueryEngine.java

```diff
@@ -148,7 +148,8 @@ public void close() throws IOException
     private final BufferAggregator[] aggregators;
     private final PositionMaintainer positionMaintainer;
 
-    private final TreeMap<ByteBuffer, Integer> positions;
+    private final Map<ByteBuffer, Integer> positions = Maps.newTreeMap();
```

Comment:
It iterates into a linked hash map later. I'm not completely clear what the bytebuffer is in this key case. I'm assuming it's the bytes comparable for the timestamp, but I'm not 100% certain. I find it odd that the treemap is ordering on the bytebuffer without having some sort of custom comparator.